bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/handler/RoombaHandler.java

```diff
@@ -0,0 +1,520 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.irobot.handler;
+
+import static org.openhab.binding.irobot.IRobotBindingConstants.*;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Hashtable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.irobot.internal.IdentProtocol;
+import org.openhab.binding.irobot.internal.IdentProtocol.IdentData;
+import org.openhab.binding.irobot.internal.MQTTProtocol;
+import org.openhab.binding.irobot.internal.RawMQTT;
+import org.openhab.binding.irobot.roomba.RoombaConfiguration;
+import org.openhab.binding.irobot.roomba.RoombaMqttBrokerConnection;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * The {@link RoombaHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author hkuhn42 - Initial contribution
+ * @author Pavel Fedin <pavel_fedin@mail.ru> - Rewrite for 900 series
+ */
+@NonNullByDefault
+public class RoombaHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(RoombaHandler.class);
+    private final ExecutorService singleThread = Executors.newSingleThreadExecutor();
```

Comment:
Because i want to be sure that all the commands are serialized and come in order, and "scheduler" doesn't guarantee that, it can execute multiple requests in parallel in different threads. And i'm not sure if HiveMQ calls are reentrant.